includes/class-wc-coupon.php

```diff
@@ -10,30 +11,53 @@
  * The WooCommerce coupons class gets coupon data from storage and checks coupon validity.
  *
  * @class 		WC_Coupon
- * @version		2.3.0
+ * @version		2.6.0
  * @package		WooCommerce/Classes
  * @category	Class
  * @author		WooThemes
  *
- * @property    string $discount_type
- * @property    string $coupon_amount
- * @property    string $individual_use
- * @property    array $product_ids
- * @property    array $exclude_product_ids
- * @property    string $usage_limit
- * @property    string $usage_limit_per_user
- * @property    string $limit_usage_to_x_items
- * @property    string $usage_count
- * @property    string $expiry_date
- * @property    string $free_shipping
- * @property    array $product_categories
- * @property    array $exclude_product_categories
- * @property    string $exclude_sale_items
- * @property    string $minimum_amount
- * @property    string $maximum_amount
- * @property    array $customer_email
  */
-class WC_Coupon {
+class WC_Coupon extends WC_Legacy_Coupon implements WC_Data {
+
+	/**
+	 * Data array, with defaults.
+	 *
+	 * When migrating to custom tables, these will be columns.
```

Comment:
I thought part of the reason (on top of making it easier for REST, CLI, etc to reuse code) was to be able to switch out CPT storage for tables at a later date -- though it sounds like that's probably only for certain tables based on your reply. So I think your proposal is good: everything in _data, custom fields can be in _meta_data.